schema/python/example/feed_generator.py

```diff
@@ -0,0 +1,623 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from datetime import datetime
+from datetime import date
+
+class IMDBExample:
+    """The IMDBExample reads the database to generate proto ItemList object from 
+    movies, series and episodes used for generating JSON-LD feed.
+    """
+    
+    def generate_feed(self,cursor, schema, serializer):
+        """Call __movies_to_proto, __series_to_proto, __episodes_to_proto and generate feed.
+
+        Args:
+            cursor (mysql.cursor): Mysql cursor which is used to query the database.
+            schema (module): The compiled protobuf schema.
+            serializer (schemaorgutils.JSONLDFeedSerializer): To serialize feed.
+
+        """
+
+        self.__movies_to_proto( cursor, schema, serializer)
```

Comment:
don't pass the serializer, instead you can use yield.
change the method names to _get_movies(cursor, schema)